### PR TITLE
Added badge evidence to award action

### PR DESCRIPTION
**Pull request by DanielBrierton**

### CODE REVIEW

File: web/public/css/footer.less [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #footer {
   position: relative;
-  z-index: 501;
+  z-index: 1002;
```
Comment (DanielBrierton) [inline web/public/css/footer.less]: z-index changes needed for handling popup actions taller than the screen. Needs to be fully tested, may need changes in cd-common

File: web/public/js/directives/cd-award-badge/template.dust [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  <div class="cd-award-badge__selection">
+    <div class="cd-award-badge__selection-select">
+      <ui-select ng-model="$ctrl.selectedBadge" on-select="$ctrl.badgeSelected()">
+        <ui-select-match allow-clear="true" placeholder="{@i18n key="Award a badge..."/}">{{ $select.selected.name  }}</ui-select-match>
+        <ui-select-choices repeat="badge in $ctrl.badges" position="up">
+          <span ng-bind-html="badge.name"></span>
+        </ui-select-choices>
+      </ui-select>
+    </div>
+    <button class="btn btn-warning btn-award-badge btn-block cd-award-badge__selection-button" ng-disabled="!$ctrl.awardBadgeButtonEnabled" ng-bootbox-confirm="{@i18n key="Are you sure you want to award this badge to the below user?"/}<br />{{ $ctrl.user.name }}" ng-bootbox-confirm-action="$ctrl.awardBadge($ctrl.selectedBadge)">{@i18n key="Award Badge"/}</button>
```
Comment (DanielBrierton) [inline web/public/js/directives/cd-award-badge/template.dust]: `$ctrl.user.name` isn't working here for some reason, needs to be investigated.

File: web/public/js/controllers/user-profile-controller.js [hunk 3]
```diff
@@ -120,6 +120,19 @@ function cdUserProfileCtrl($scope, $rootScope, $state, $window, auth, cdUsersSer
...
 
+  $scope.showBadgeModal = function (badge) {
+    var badgeModal = $uibModal.open({
+      controller: function () {
+        this.badge = badge;
+        this.close = badgeModal.close;
+      },
+      bindToController: true,
+      controllerAs: '$ctrl',
+      template: '<div><cd-badge-cert badge="$ctrl.badge" /><div class="modal-footer"><button ng-click="$ctrl.close()" class="btn btn-default pull-right">' + $translate.instant('Close') + '</button></div>',
```
Comment (DanielBrierton) [inline web/public/js/controllers/user-profile-controller.js]: Maybe export this to a template

Comment (Wardormeur) [inline web/public/js/controllers/user-profile-controller.js]: I believe it's fine for a popup that is 2 lines

File: web/public/js/directives/cd-action-bar/cd-action-bar-popup-item/style.less [hunk 4]
```diff
@@ -35,27 +37,36 @@
...
-      }
-
-      &--mobile {
-        display: none;
-        left: 0;
-        width: 100%;
+      &--visible {
+        display: block;
 
         @media (max-width: @screen-xs-max) {
```
Comment (Wardormeur) [inline web/public/js/directives/cd-action-bar/cd-action-bar-popup-item/style.less]: this is weird, mind xplaining?
is it that you switch from __popup to __full-width-popup when the screen is xs?

File: web/public/js/directives/cd-action-bar/index.js [hunk 5]
```diff
@@ -38,6 +45,16 @@
           $footer.off('inview');
           $element.remove();
         };
+
+        ctrl.$onChanges = function (changes) {
+          if (changes.forceFixed) {
```
Comment (Wardormeur) [inline web/public/js/directives/cd-action-bar/index.js]: I guess it'll not trigger if changes.forceFixed isn't ?
It means you can't null it ?

File: web/public/js/directives/cd-badge-cert/template.dust [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<div class="cd-badge-cert">
+  <h2>{{ $ctrl.badge.name }}</h2>
+  <div class="cd-badge-cert__info">
+    <div class="cd-badge-cert__image">
+      <img ng-src="{{ $ctrl.badge.imageUrl }}" />
+    </div>
+    <p class="cd-badge-cert__description" ng-bind-html="$ctrl.badge.consumerDescription | linky"></p>
```
Comment (Wardormeur) [inline web/public/js/directives/cd-badge-cert/template.dust]: holy, didn't know abt linky, wizardry !

Comment (Wardormeur) [inline web/public/js/directives/cd-badge-cert/template.dust]: is consumerDescription manual user entry? If yes, ng-bind-html could be dangerous

File: web/public/js/directives/cd-award-badge/index.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    .module('cpZenPlatform')
+    .component('cdAwardBadge', {
+      bindings: {
+        user: '<'
+      },
+      controller: ['cdBadgesService', 'usSpinnerService', '$translate', 'alertService', '$rootScope', function (cdBadgesService, usSpinnerService, $translate, alertService, $rootScope) {
+        var ctrl = this;
+        ctrl.evidence = '';
+
+        ctrl.badgeSelection = ctrl.badgeSelection || {};
```
Comment (Wardormeur) [inline web/public/js/directives/cd-award-badge/index.js]: why is badgeSelection not bound ?

**Comment by Wardormeur** [comment]
Cookie policy covers the toolbar :D

**Comment by Wardormeur** [comment]
When having a list of users quite limited, the popup covers the navigation bar. I think it'd be better if if was anchored to the navigation bar instead

**Comment by Wardormeur** [comment]
I'm nearly tempted to say an "X" to close wouldn't be too much, considering the size of it. Or a slightly more dark color to differenciate the action bar from the popup (i know it's already the case :))

File: web/public/js/directives/cd-badge-cert/template.dust (hunk 6 above)
Comment (DanielBrierton) [inline web/public/js/directives/cd-badge-cert/template.dust]: consumerDescription is user entry into badgekit, so at the moment, we're in full control. I'd imagine we'd moderate badges even if we opened up the ability to create badges?

File: web/public/js/directives/cd-award-badge/index.js (hunk 7 above)
Comment (DanielBrierton) [inline web/public/js/directives/cd-award-badge/index.js]: Possible leftover from previous implementation, will check and probably get rid of it.

File: web/public/js/directives/cd-action-bar/cd-action-bar-popup-item/style.less (hunk 4 above)
Comment (DanielBrierton) [inline web/public/js/directives/cd-action-bar/cd-action-bar-popup-item/style.less]: Gonna add comments on this

File: web/public/js/directives/cd-action-bar/index.js (hunk 5 above)
Comment (DanielBrierton) [inline web/public/js/directives/cd-action-bar/index.js]: `changes.forceFixed` is an object from Angular. If `forceFixed` has been changed, `changes.forceFixed` will always be an object. `changes.forceFixed.currentValue` would be null, not `changes.forceFixed`.

**Pull request merged**